pkg/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -15,21 +15,890 @@
+						},
+					},
+					{
+						Direction: networkpolicy.DirectionOut,
+						To: networkpolicy.NetworkPolicyPeer{
+							AddressGroups: []string{getNormalizedUID(toGroupSelector("nsA", &selectorB, nil).NormalizedName)},
+						},
+					},
+				},
+				AppliedToGroups: []string{getNormalizedUID(toGroupSelector("nsA", &metav1.LabelSelector{}, nil).NormalizedName)},
+			},
+			expAppliedToGroups: 1,
+			expAddressGroups:   3,
+		},
+		{
+			name: "update-egress-rule-selector",
+			updatedNetworkPolicy: &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "npA", UID: "uidA"},
+				Spec: networkingv1.NetworkPolicySpec{
+					PodSelector: metav1.LabelSelector{},
+					Ingress: []networkingv1.NetworkPolicyIngressRule{
+						{
+							From: []networkingv1.NetworkPolicyPeer{
+								{
+									PodSelector:       &selectorB,
+									NamespaceSelector: &selectorC,
+								},
+							},
+						},
+					},
+					Egress: []networkingv1.NetworkPolicyEgressRule{
+						{
+							To: []networkingv1.NetworkPolicyPeer{
+								{
+									PodSelector: &selectorA,
+								},
+							},
+						},
+					},
+				},
+			},
+			expNetworkPolicy: &antreatypes.NetworkPolicy{
+				UID:       "uidA",
+				Name:      "npA",
+				Namespace: "nsA",
+				Rules: []networkpolicy.NetworkPolicyRule{
+					{
+						Direction: networkpolicy.DirectionIn,
+						From: networkpolicy.NetworkPolicyPeer{
+							AddressGroups: []string{getNormalizedUID(toGroupSelector("nsA", &selectorB, &selectorC).NormalizedName)},
+						},
+					},
+					{
+						Direction: networkpolicy.DirectionOut,
+						To: networkpolicy.NetworkPolicyPeer{
+							AddressGroups: []string{getNormalizedUID(toGroupSelector("nsA", &selectorA, nil).NormalizedName)},
+						},
+					},
+				},
+				AppliedToGroups: []string{getNormalizedUID(toGroupSelector("nsA", &metav1.LabelSelector{}, nil).NormalizedName)},
+			},
+			expAppliedToGroups: 1,
+			expAddressGroups:   2,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, npc := newController()
+			npc.addNetworkPolicy(oldNP)
+			npc.updateNetworkPolicy(oldNP, tt.updatedNetworkPolicy)
+			key, _ := keyFunc(oldNP)
+			actualPolicyObj, _, _ := npc.internalNetworkPolicyStore.Get(key)
+			actualPolicy := actualPolicyObj.(*antreatypes.NetworkPolicy)
+			if actualAppliedToGroups := len(npc.appliedToGroupStore.List()); actualAppliedToGroups != tt.expAppliedToGroups {
+				t.Errorf("updateNetworkPolicy() got %v, want %v", actualAppliedToGroups, tt.expAppliedToGroups)
+			}
+			if actualAddressGroups := len(npc.addressGroupStore.List()); actualAddressGroups != tt.expAddressGroups {
+				t.Errorf("updateNetworkPolicy() got %v, want %v", actualAddressGroups, tt.expAddressGroups)
+			}
+			if !reflect.DeepEqual(actualPolicy, tt.expNetworkPolicy) {
+				t.Errorf("updateNetworkPolicy() got %#v, want %#v", actualPolicy, tt.expNetworkPolicy)
+			}
+		})
+	}
+}
+
+func TestAddPod(t *testing.T) {
```

Comment:
Do you plan to update the following tests to the style same as above?

Reply:
yes i added the WIP to use same style for this test case too

Reply:
updated in new commit.. so you can see delta...